master/internal/rm/kubernetesrm/resource_pool_intg_test.go

```diff
@@ -0,0 +1,301 @@
+
+				// ResourcesReleased
+				rp.ResourcesReleased(sproto.ResourcesReleased{
+					AllocationID: n,
+					ResourcePool: rp.poolConfig.PoolName,
+				})
+				req, ok = rp.reqList.TaskByID(n)
+				require.False(t, ok)
+				require.Nil(t, req)
+				require.Empty(t, rp.allocationIDToContainerID[n])
+				require.Empty(t, rp.allocationIDToJobID[n])
+			}
+		})
+	}
+}
+
+func TestUpdatePodStatus(t *testing.T) {
+	cases := []struct {
+		name           string
+		state          sproto.SchedulingState
+		containerMatch bool
+		runningPod     int
+	}{
+		{"scheduled", sproto.SchedulingStateScheduled, true, 1},
+		{"backfilled", sproto.SchedulingStateScheduledBackfilled, true, 1},
+		{"queued", sproto.SchedulingStateQueued, true, 0},
+		{"container error", sproto.SchedulingStateScheduled, false, 0},
+	}
+	for _, tt := range cases {
+		t.Run(tt.name, func(t *testing.T) {
+			rp, jobID := testResourcePoolWithJob(t, defaultSlots)
+			newAllocID := newTestPod(t, rp, jobID)
+
+			// Check that the allocation is queued first.
+			require.Equal(t, int(sproto.SchedulingStateQueued), rp.allocationIDToRunningPods[newAllocID])
+			require.Zero(t, rp.allocationIDToRunningPods[newAllocID])
+
+			req := sproto.UpdatePodStatus{
+				ContainerID: "bogus",
+				State:       tt.state,
+			}
+			if tt.containerMatch {
+				req.ContainerID = string(rp.allocationIDToContainerID[newAllocID])
+			}
+			rp.UpdatePodStatus(req)
+
+			// If scheduled (backfilled or scheduled), check that running pods++
+			require.Equal(t, tt.runningPod, rp.allocationIDToRunningPods[newAllocID])
+		})
+	}
+}
+
+func TestPendingPreemption(t *testing.T) {
+	rp, jobID := testResourcePoolWithJob(t, defaultSlots)
+	allocID := newTestPod(t, rp, jobID)
+	err := rp.PendingPreemption(sproto.PendingPreemption{
+		AllocationID: allocID,
+	})
+	require.Equal(t, rmerrors.ErrNotSupported, err)
+}
+
+func TestSetGroupWeight(t *testing.T) {
+	rp, _ := testResourcePoolWithJob(t, defaultSlots)
+	err := rp.SetGroupWeight(sproto.SetGroupWeight{})
+	require.Equal(t, rmerrors.UnsupportedError("set group weight is unsupported in k8s"), err)
+}
+
+func TestSetGroupPriority(t *testing.T) {
+	cases := []struct {
+		name        string
+		newPriority int
+	}{
+		{"no change", int(config.KubernetesDefaultPriority)},
+		{"increase", 100},
+		{"decrease", 1},
+		{"negative", -10}, // doesn't make sense, but it is allowed
+	}
+	for _, tt := range cases {
+		t.Run(tt.name, func(t *testing.T) {
+			rp, jobID := testResourcePoolWithJob(t, defaultSlots)
+			group := rp.getOrCreateGroup(jobID)
+			require.Equal(t, int(config.KubernetesDefaultPriority), *group.Priority)
+
+			err := rp.SetGroupPriority(sproto.SetGroupPriority{
+				Priority:     tt.newPriority,
+				ResourcePool: rp.poolConfig.PoolName,
+				JobID:        jobID,
+			})
+
+			require.NoError(t, err)
+			require.Equal(t, tt.newPriority, *rp.getOrCreateGroup(jobID).Priority)
+		})
+	}
+}
+
+func TestRecoverJobPosition(t *testing.T) {
```

Comment:
just dont worry about this test since we're getting rid of move job.

Reply:
I'll remove it